### PR TITLE
Allow array of patterns for backend pattern

### PR DESCRIPTION
**Pull request by nekuz0r**
Allow to have multiple patterns to match the same backend.

### CODE REVIEW

**Comment by cliftonc** [comment]
Looks like a good feature, thanks for contributing, can you add a test and a comment in the readme?

File: src/middleware/backend.js [hunk 1]
```diff
@@ -15,7 +15,11 @@ module.exports = function(config)  {
   return function selectBackend(req, res, next) {
     if (config.backend) {
       req.backend = _.find(config.backend, function(server) {
-          if (server.pattern) { return new RegExp(server.pattern).test(req.url); }
+          if (server.pattern) {
+            return [].concat(server.pattern).reduce(function (matched, pattern) {
```
Comment (codeaholics) [inline src/middleware/backend.js]: Could use Array.some here instead of reduce with the matched state.

**Comment by nekuz0r** [comment]
@cliftonc Sure, will add tests and update the documentation
@codeaholics Yes, didn't thought about `Array.some`, will be more optimized :)

**Comment by cliftonc** [comment]
@nekuz0r need any assistance with this - anything we can do to help?

**Comment by nekuz0r** [comment]
@cliftonc Sorry, didn't had time to check how to write that test so far.
Is adding a new backend with array of pattern in test config + handler in `stubServer.js` and adding a test for it in `compoxure.test.js` enough ?

**Comment by cliftonc** [comment]
Yes, that's right - we're keeping all the tests together until it becomes un-manageable and we break it out by type.

**Comment by nekuz0r** [comment]
@cliftonc Test added, commits squashed, branch rebased

**Comment by cliftonc** [comment]
lgtm - thanks!

**Comment by cliftonc** [comment]
Released with 0.3.81

**Pull request merged**